app/controllers/SeriesDetailsController.scala

```diff
@@ -19,33 +25,59 @@ class SeriesDetailsController @Inject()(val controllerComponents: SecurityCompon
 
   private val secureAction = Secure("OidcClient")
-  private val client = graphqlConfiguration.getClient[Data, Variables]()
+  private val getSeriesClient = graphqlConfiguration.getClient[getSeries.Data, getSeries.Variables]()
+  private val addConsignmentClient = graphqlConfiguration.getClient[addConsignment.Data, addConsignment.Variables]()
 
   val selectedSeriesForm = Form(
     mapping(
-      "seriesId" -> nonEmptyText
+      "series" -> nonEmptyText
     )(SelectedSeriesData.apply)(SelectedSeriesData.unapply)
   )
 
-  def seriesDetails(): Action[AnyContent] = secureAction.async { implicit request: Request[AnyContent] =>
-    val userTransferringBody = keycloakConfiguration.verifyToken(request.token.getValue)
-      .map(t => t.getOtherClaims.get("body").asInstanceOf[String])
-    val variables: Variables = new Variables(userTransferringBody)
+  private def getSeriesDetails(request: Request[AnyContent], status: Status, form: Form[SelectedSeriesData])(implicit requestHeader: RequestHeader) = {
+    val userTransferringBody: Option[String] = keycloakConfiguration.verifyToken(request.token.getValue)
+      .flatMap(t => t.getOtherClaims.asScala.get("body").asInstanceOf[Option[String]])
+    val variables: getSeries.Variables = new getSeries.Variables(userTransferringBody)
 
-    client.getResult(request.token, document, Some(variables)).map(data => {
+    getSeriesClient.getResult(request.token, getSeries.document, Some(variables)).map(data => {
       if (data.data.isDefined) {
         val seriesData: Seq[(String, String)] = data.data.get.getSeries.map(s => (s.seriesid.toString, s.code.getOrElse("")))
-        Ok(views.html.seriesDetails(seriesData, selectedSeriesForm))
+        status(views.html.seriesDetails(seriesData, form))
       } else {
-        Ok(views.html.error(data.errors.map(e => e.message).mkString))
+        status(views.html.error(data.errors.map(e => e.message).mkString))
       }
     })
+  }
 
+  def seriesDetails(): Action[AnyContent] = secureAction.async { implicit request: Request[AnyContent] =>
+    getSeriesDetails(request, Ok, selectedSeriesForm)
   }
 
-  // Submit returns current page until next page is ready
-  def seriesSubmit(): Action[AnyContent] =  secureAction { implicit request: Request[AnyContent] =>
-    Redirect(routes.SeriesDetailsController.seriesDetails())
+  def seriesSubmit(): Action[AnyContent] =  secureAction.async { implicit request: Request[AnyContent] =>
+    val formValidationResult: Form[SelectedSeriesData] = selectedSeriesForm.bindFromRequest
+
+    val errorFunction: Form[SelectedSeriesData] => Future[Result] = { formWithErrors: Form[SelectedSeriesData] =>
+      getSeriesDetails(request, BadRequest, formWithErrors)
+    }
+
+    val successFunction: SelectedSeriesData => Future[Result] = { formData: SelectedSeriesData =>
+      val userId = keycloakConfiguration.verifyToken(request.token.getValue)
+        .map(t => t.getOtherClaims.get("user_id").asInstanceOf[String])
```

Comment:
Might be worth extracting this logic and the transferring body logic to another module, so that the token parsing is reusable and in one place.